sa/model_test.go

```diff
@@ -224,3 +234,86 @@ func TestPopulateAttemptedFieldsBadJSON(t *testing.T) {
 		})
 	}
 }
+
+func TestCerficatesTableContainsDuplicateSerials(t *testing.T) {
+	testCtx := setup(t)
+	defer testCtx.cleanUp()
+
+	serialString := core.SerialToString(big.NewInt(1337))
+
+	// Insert a certificate with a serial of `1337`.
+	testCtx.insertCertificate(t, "1337.com", 1337, 1)
+
+	// This should return the certificate that we just inserted.
+	_, err := SelectCertificate(testCtx.db, serialString)
+	test.AssertNotError(t, err, "received an error for a valid query")
+
+	// Insert a certificate with a serial of `1337` but for a different
+	// hostname.
+	testCtx.insertCertificate(t, "1337.net", 1337, 1)
+
+	// With a duplicate being present, this should error.
+	_, err = SelectCertificate(testCtx.db, serialString)
+	test.AssertError(t, err, "should've received an error for multiple rows")
+}
+
+type testCtx struct {
+	db      *db.WrappedMap
+	ssa     *SQLStorageAuthority
+	cleanUp func()
+	fc      clock.FakeClock
+}
+
+func (ctx testCtx) insertCertificate(t *testing.T, hostname string, serial, regID int64) {
+	serialBigInt := big.NewInt(serial)
+	serialString := core.SerialToString(serialBigInt)
+
+	rawCert := x509.Certificate{
```

Comment:
Technically this is not a `rawCert`, but a `template` (that is what the corresponding param to CreateCertificate is called).